### PR TITLE
Add blockchain test linting

### PR DESCRIPTION
**Pull request by jwasinger**
**Status:** Work in progress
**Depends On:** https://github.com/ethereum/tests/pull/275 
Adding linting for blockchain fillers and generated tests will help to prevent issues like https://github.com/ethereum/cpp-ethereum/issues/4369

### CODE REVIEW

File: JSONSchema/bc-schema.json [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+                "transactions": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/Transaction"
+                    }
+                },
+                "uncleHeaders": {
+                    "type": "array"
+                },
+                "expectExceptionHomestead": {
```
Comment (winsvega) [inline JSONSchema/bc-schema.json]: this field should be only in fillers

File: JSONSchema/bc-schema.json [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+                "expectExceptionALL": {
+                    "type": "string"
+                },
+                "expectExceptionByzantium": {
+                    "type": "string"
+                },
+                "expectExceptionConstantinople": {
+                    "type": "string"
+                },
+                "blocknumber": {
```
Comment (winsvega) [inline JSONSchema/bc-schema.json]: this field is optional but also could be forced to happen only in fillers

File: JSONSchema/bc-schema.json [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+                    "$ref": "#/definitions/IntegerOrEmptyOrConfusedHex"
+                },
+                "chainname": {
```
Comment (winsvega) [inline JSONSchema/bc-schema.json]: optional

File: JSONSchema/bc-schema.json [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+                    "type": "string"
+                },
+                "chainnetwork": {
```
Comment (winsvega) [inline JSONSchema/bc-schema.json]: optional

File: JSONSchema/bc-schema.json (hunk 1 above)
Comment (jwasinger) [reply JSONSchema/bc-schema.json]: I'm seeing it in quite a lot of filled tests.

**Comment by jwasinger** [comment]
@winsvega right now all block fields are optional in the schema because there are tests with and without them.

**Comment by winsvega** [comment]
the optional fields in blockchain test are not needed in the final test. the client should be able to input a block by its RLP. actually those fields are confusing client developers so we should remove it from filled tests.

@pirapira

**Pull request merged**